src/api/v4/error.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * Build a cartoError from a generic error
+ * @constructor
+ * 
+ * @return {CartoError} A well formed object representing the error.
+ */
+function CartoError (error) {
+  if (!error) {
+    this.message = 'unexpected error';
+    return this;
+  }
+  if (_isWindshaftError(error)) {
+    this.message = error.errors[0];
+    return this;
+  }
+  if (error.message) {
+    this.message = error.message;
+    return this;
+  }
+  this.message = 'unexpected error';
+}
+
+function _isWindshaftError (error) {
+  return error && error.errors_with_context;
+}
+
+module.exports = CartoError;
+
+/**
+ * 
+ * Represents an error in the carto library.
+ * 
+ * Some actions like adding a layer to a map will trigger a **reload cycle**
```

Comment:
"reload cycle" is not very clear to me. I'd try to find something more meaningful for developers.

Reply:
umm, any suggestion?

I would try to explain how carto.js works (including the reload-cycle) to developers in order to make it easier to understand